api/server/router/system/system_routes.go

```diff
@@ -90,24 +90,45 @@ func (s *systemRouter) getVersion(ctx context.Context, w http.ResponseWriter, r
 }
 
 func (s *systemRouter) getDiskUsage(ctx context.Context, w http.ResponseWriter, r *http.Request, vars map[string]string) error {
+	if err := httputils.ParseForm(r); err != nil {
+		return err
+	}
+
+	var objectTypes map[types.DiskUsageObject]struct{}
```

Comment:
Nit, it seems like you really want a `map[string]bool` here.

Reply:
Well, I want a set of enums, and the most idiomatic way to do that in Go is using a `struct{}`-valued map AFAIK.
Making this a map of `bool`s would introduce ambiguity - i.e. what does `containers`: `false` mean? If it should not be computed, it should just not be there.
Also, we get basically just an over-complicated `struct` with booleans, so, again, might as well just use a struct instead.